src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -312,100 +318,164 @@ def get_dependencies(cls, mgr: "CephadmOrchestrator",
     def __init__(self, mgr: "CephadmOrchestrator"):
         self.mgr: "CephadmOrchestrator" = mgr
 
-    def get_self_signed_certificates_with_label(self, svc_spec: ServiceSpec, daemon_spec: CephadmDaemonDeploySpec, label: str) -> CertKeyPair:
+    def get_self_signed_certificates_with_label(self, svc_spec: ServiceSpec, daemon_spec: CephadmDaemonDeploySpec, label: str) -> TLSCredentials:
         svc_name = svc_spec.service_name()
         ip = self.mgr.inventory.get_addr(daemon_spec.host)
         host_fqdn = self.mgr.get_fqdn(daemon_spec.host)
-        tls_pair = self.mgr.cert_mgr.get_self_signed_cert_key_pair(svc_name, host_fqdn, label)
-        if not tls_pair:
-            tls_pair = self.mgr.cert_mgr.generate_cert(host_fqdn, ip)
-            self.mgr.cert_mgr.save_self_signed_cert_key_pair(svc_name, tls_pair, host=daemon_spec.host, label=label)
-        return tls_pair
+        tls_creds = self.mgr.cert_mgr.get_self_signed_cert_key_pair(svc_name, host_fqdn, label)
+        if not tls_creds:
+            tls_creds = self.mgr.cert_mgr.generate_cert(host_fqdn, ip)
+            self.mgr.cert_mgr.save_self_signed_cert_key_pair(svc_name, tls_creds, host=daemon_spec.host, label=label)
+        return tls_creds
 
     def get_certificates(self,
                          daemon_spec: CephadmDaemonDeploySpec,
                          ips: List[str] = [],
                          fqdns: List[str] = [],
-                         custom_sans: List[str] = []
-                         ) -> CertKeyPair:
+                         custom_sans: List[str] = [],
+                         ca_cert_required: bool = False
+                         ) -> TLSCredentials:
 
         svc_spec = cast(ServiceSpec, self.mgr.spec_store[daemon_spec.service_name].spec)
         if not self.requires_certificates or not svc_spec.ssl:
-            return EMPTY_TLS_KEYPAIR
+            return EMPTY_TLS_CREDENTIALS
+
+        return self.get_certificates_generic(
+            svc_spec=svc_spec,
+            daemon_spec=daemon_spec,
+            cert_attr='ssl_cert',
+            key_attr='ssl_key',
+            cert_source_attr='certificate_source',
+            cert_name=self.cert_name,
+            key_name=self.key_name,
+            ips=ips,
+            fqdns=fqdns,
+            custom_sans=custom_sans,
+            ca_cert_attr='ssl_ca_cert' if ca_cert_required else None,
+            ca_cert_name=self.ca_cert_name if ca_cert_required else None,
```

Comment:
just for readability: can we group params that are related to attributes and names on consecutive lines?